Search Guard 6.4.1 or 6.4.2 not compatible

### Problem description
Can't install Search Guard if we change the version into .env to 6.4.1 or 6.4.2

### Extra information

This is because we hardcoded the versions from elasticsearch/Dockerfile and kibana/Dockerfile, while Search Guard has versions metrix
https://docs.search-guard.com/latest/search-guard-versions

Solution to move SG_VERSION to .env file

#### Docker logs

```
Step 7/7 : RUN elasticsearch-plugin install --batch com.floragunn:search-guard-6:${SG_VERSION} 	&& chmod +x plugins/search-guard-6/tools/*.sh 	&& chown -R elasticsearch config/sg/ 	&& chmod -R go= config/sg/
 ---> Running in 93c8397bc756
-> Downloading com.floragunn:search-guard-6:6.4.2-23.0 from maven central
Exception in thread "main" java.io.FileNotFoundException: https://repo1.maven.org/maven2/com/floragunn/search-guard-6/6.4.2-23.0/search-guard-6-6.4.2-23.0.zip
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:488)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1963)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1958)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1957)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1525)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1509)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:245)
	at org.elasticsearch.plugins.InstallPluginCommand.downloadZip(InstallPluginCommand.java:380)
	at org.elasticsearch.plugins.InstallPluginCommand.downloadAndValidate(InstallPluginCommand.java:453)
	at org.elasticsearch.plugins.InstallPluginCommand.download(InstallPluginCommand.java:264)
	at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:229)
	at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:216)
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
	at org.elasticsearch.cli.MultiCommand.execute(MultiCommand.java:77)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:124)
	at org.elasticsearch.cli.Command.main(Command.java:90)
	at org.elasticsearch.plugins.PluginCli.main(PluginCli.java:47)
Caused by: java.io.FileNotFoundException: https://repo1.maven.org/maven2/com/floragunn/search-guard-6/6.4.2-23.0/search-guard-6-6.4.2-23.0.zip
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1909)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1509)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(HttpURLConnection.java:3033)
	at java.base/java.net.URLConnection.getHeaderFieldLong(URLConnection.java:636)
	at java.base/java.net.URLConnection.getContentLengthLong(URLConnection.java:508)
	at java.base/java.net.URLConnection.getContentLength(URLConnection.java:492)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getContentLength(HttpsURLConnectionImpl.java:389)
	at org.elasticsearch.plugins.InstallPluginCommand.downloadZip(InstallPluginCommand.java:379)
	... 10 more
ERROR: Service 'elasticsearch' failed to build: The command '/bin/sh -c elasticsearch-plugin install --batch com.floragunn:search-guard-6:${SG_VERSION} 	&& chmod +x plugins/search-guard-6/tools/*.sh 	&& chown -R elasticsearch config/sg/ 	&& chmod -R go= config/sg/' returned a non-zero code: 1
```

--- a/.env
+++ b/.env
@@ -1,1 +1,9 @@
 ELK_VERSION=6.4.0
+
+# Check Search Guard versions matrix at https://docs.search-guard.com/latest/search-guard-versions
+# or the Maven Central repository:
+#  * https://search.maven.org/search?q=g:com.floragunn%20AND%20a:search-guard-6&core=gav
+#  * https://search.maven.org/search?q=g:com.floragunn%20AND%20a:search-guard-kibana-plugin&core=gav
+#
+SG_VERSION=23.1
+SG_VERSION_KIBANA=15

--- a/.env
+++ b/.env
@@ -1,4 +1,4 @@
-ELK_VERSION=6.4.0
+ELK_VERSION=6.4.2
 
 # Check Search Guard versions matrix at https://docs.search-guard.com/latest/search-guard-versions
 # or the Maven Central repository:

--- a/README.md
+++ b/README.md
@@ -1,7 +1,7 @@
 # Elastic stack (ELK) on Docker
 
 [![Join the chat at https://gitter.im/deviantony/docker-elk](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/deviantony/docker-elk?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
-[![Elastic Stack version](https://img.shields.io/badge/ELK-6.4.0-blue.svg?style=flat)](https://github.com/deviantony/docker-elk/issues/312)
+[![Elastic Stack version](https://img.shields.io/badge/ELK-6.4.2-blue.svg?style=flat)](https://github.com/deviantony/docker-elk/issues/332)
 [![Build Status](https://api.travis-ci.org/deviantony/docker-elk.svg?branch=searchguard)](https://travis-ci.org/deviantony/docker-elk)
 
 Run the latest version of the [Elastic stack](https://www.elastic.co/elk-stack) with Docker and Docker Compose.
@@ -142,7 +142,7 @@ Authenticate against Kibana:
 $ curl -XPOST -D- 'http://localhost:5601/api/v1/auth/login' \
     -c /tmp/sg_cookies \
     -H 'Content-Type: application/json' \
-    -H 'kbn-version: 6.4.0' \
+    -H 'kbn-version: 6.4.2' \
     -d '{"username":"kibanaro","password":"kibanaro"}'
 ```
 
@@ -152,7 +152,7 @@ Create an index pattern via the Kibana API:
 $ curl -XPOST -D- 'http://localhost:5601/api/saved_objects/index-pattern' \
     -b /tmp/sg_cookies \
     -H 'Content-Type: application/json' \
-    -H 'kbn-version: 6.4.0' \
+    -H 'kbn-version: 6.4.2' \
     -d '{"attributes":{"title":"logstash-*","timeFieldName":"@timestamp"}}'
 ```
 

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -7,6 +7,7 @@ services:
       context: elasticsearch/
       args:
         ELK_VERSION: $ELK_VERSION
+        SG_VERSION: $SG_VERSION
     volumes:
       - ./elasticsearch/config/elasticsearch.yml:/usr/share/elasticsearch/config/elasticsearch.yml:ro
     ports:
@@ -40,6 +41,7 @@ services:
       context: kibana/
       args:
         ELK_VERSION: $ELK_VERSION
+        SG_VERSION_KIBANA: $SG_VERSION_KIBANA
     volumes:
       - ./kibana/config/:/usr/share/kibana/config:ro
     ports:

--- a/elasticsearch/Dockerfile
+++ b/elasticsearch/Dockerfile
@@ -7,11 +7,9 @@ COPY config/sg/ config/sg/
 COPY bin/ bin/
 
 # Search Guard plugin
-# https://docs.search-guard.com/latest/search-guard-versions
-# https://search.maven.org/search?q=g:com.floragunn%20AND%20a:search-guard-6&core=gav
 ARG ELK_VERSION
-ENV SG_VERSION=${ELK_VERSION}-23.0
-RUN elasticsearch-plugin install --batch com.floragunn:search-guard-6:${SG_VERSION} \
+ARG SG_VERSION
+RUN elasticsearch-plugin install --batch com.floragunn:search-guard-6:${ELK_VERSION}-${SG_VERSION} \
 	&& chmod +x plugins/search-guard-6/tools/*.sh \
 	&& chown -R elasticsearch config/sg/ \
 	&& chmod -R go= config/sg/

--- a/kibana/Dockerfile
+++ b/kibana/Dockerfile
@@ -4,11 +4,9 @@ ARG ELK_VERSION
 FROM docker.elastic.co/kibana/kibana-oss:${ELK_VERSION}
 
 # Search Guard plugin
-# https://docs.search-guard.com/latest/search-guard-versions
-# https://search.maven.org/search?q=g:com.floragunn%20AND%20a:search-guard-kibana-plugin&core=gav
 ARG ELK_VERSION
-ENV SG_VERSION=${ELK_VERSION}-14
-RUN kibana-plugin install https://repo1.maven.org/maven2/com/floragunn/search-guard-kibana-plugin/${SG_VERSION}/search-guard-kibana-plugin-${SG_VERSION}.zip
+ARG SG_VERSION_KIBANA
+RUN kibana-plugin install https://repo1.maven.org/maven2/com/floragunn/search-guard-kibana-plugin/${ELK_VERSION}-${SG_VERSION_KIBANA}/search-guard-kibana-plugin-${ELK_VERSION}-${SG_VERSION_KIBANA}.zip
 
 # Add your kibana plugins setup here
 # Example: RUN kibana-plugin install <name|url>

